MODEL-GUIDE.md

```diff
@@ -301,6 +303,79 @@ The build script will enforce the following keyword conventions:
         value: 0.0.0.0
     ```
 
+* x-field-uid
+
+  * Field Unique Identifiers (UIDs) are required for every property/enum/included object.  
+  * They need to be unique (within the object), and once assigned can never be changed/re-used, even if the field itself is deprecated and eventually removed.  
+  * They are required for maintaining protobuff backwards compatibility.  
+  * Should follow protobuff [Assigning Field Numbers](https://developers.google.com/protocol-buffers/docs/proto3#assigning_field_numbers) conventions.
+
+  * Use-Case : Adding a New Property to an Existing Object
+    * Adding a new property 'peer_name' to an existing object.
+    * `x-field-uid` must be added with different value (e.g. 2)
+
+    <img src="images/use_case_1.PNG" alt="drawing" width="500"/>
+
+  * Use-Case : Adding an enum value
+    * enum must be replaced with `x-enum` 
+    * Adding a new enum `fiber` to the existing property field (`media`)
+    * `x-field-uid` must be added with different value (e.g. 2)
+    * The bundler will populate `enum` at the time of bundling. So, it will follow OpenAPI standard. 
+    * `x-enum` remain within final yaml file to generate unique proto uid.
+
+    <img src="images/use_case_2.PNG" alt="drawing" width="500"/>
+
+  * Use-Case : Including a Pre-Existing Object
+    * `x-include` to include only the field of the property (`..#/components/schemas/GlobalObject/properties/name`)
+    * Same property name (`uniquename`) must be used for both base and derived property. 
+    * The bundler will correctly merge the `x-include` (e.g. `type` and `pattern`) and drop the `x-include` keyword from the merged object.
+    * At the time of merge it will not include base `x-field-uid` `x-field-uid: 2` should present in the derived property.
+    
+    <img src="images/use_case_3.PNG" alt="drawing" width="500"/>
+
+  * Use-Case : Deprecating a Property 
+    * Model should maintain `x-field-uid: 1` when property will deprecate 
+    * `x-status` must be added with status deprecated
+
+    <img src="images/use_case_4.PNG" alt="drawing" width="500"/>
+
+  * Use-Case : Removing a Property
```

Comment:
in general on Removes, question remains, do we want things to go from 

property active -> deprecated -> removed?  or allow
property active -> removed?

the distinction here is I suspect if we go directly from property active-> removed, it implies the potential of an SDK breakeage, though serialization compatibility is maintained.  

we should clearly specify our expected lifecycle here.

Reply:
Agree.  My vote would be that we go from active --> deprecated --> removed ONLY and that we clearly document it in the guide.